### PR TITLE
test match profiling

### PR DESCRIPTION
**Pull request by havardpe**
@geirst please review

### CODE REVIEW

File: tests/search/query_profiling/query_profiling.rb [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  
+  def setup
+    @match_tag = "match_profiling"
+    @first_phase_tag = "first_phase_profiling"
+    @second_phase_tag = "second_phase_profiling"
+    set_owner('havardpe')
+    set_description("Test query profiling (matching/ranking)")
+  end
+
+  def test_rank_profiling
```
Comment (geirst) [inline tests/search/query_profiling/query_profiling.rb:16]: Consider rename to test_query_profiling.

File: tests/search/query_profiling/query_profiling.rb [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  def get_query_trace(query)
+    result = search(query).json
+    trace = result["trace"]["children"][1]["children"][0]["children"][1]["message"][0]["traces"][0]
+    assert_equal(trace["tag"], "query_execution")
+    return trace
+  end
+  
+  def verify_match_profiling(depth)
+    puts "test case: match(#{depth})"
+    verify_match_trace(get_query_trace("/search/?query=sddocname:test&format=json&hits=1&tracelevel=1&trace.profileDepth=#{depth}&type=all"),
```
Comment (geirst) [inline tests/search/query_profiling/query_profiling.rb:36]: Consider adding a function that builds the query string, avoiding duplication of the parts that are not relevant for the test cases. Now I had to scan all query strings (looking for the actual differences), making the test cases harder to read. Also, the "/search/" prefix should not be needed when calling search(query).

**Review by geirst — DISMISSED** [review]
👍

**Comment by havardpe** [comment]
@geirst PTAL

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**